### PR TITLE
Snapshot management blog

### PR DESCRIPTION
**Pull request by kolchfa-aws**
### Description
Adds a blog on Snapshot Management.
 
### Issues Resolved
none

### Check List
- [x ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by krisfreedain** [comment]
@kolchfa-aws @natebower just wanted to confirm this got through the editorial process before I publish. Please let me know - thanks

**Comment by kolchfa-aws** [comment]
@krisfreedain - This is not ready to be merged yet. It needs to go through the tech review and editorial. I can put it in draft if it's easier

**Comment by krisfreedain** [comment]
thanks for letting me know - I'll leave it until I hear from you again. thanks

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and let me know if you have any questions. Thanks!

File: _posts/snapshot-management.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+---
+layout: post
+title:  "Snapshot Management: A new way to automate snapshots"
+authors:
+-  
+date: 2022-09-16
+categories:
+ - technical-post
+
+excerpt: "Snapshot Management is a way to automatically take snapshots of your cluster's indexes and state. This is useful for restoring indexes from a snapshot after a failure or migration. With Snapshot Management you can set a schedule for creating and deleting snapshots and a list of indexes to work on. You can also get notified of snapshot activity."
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _posts/snapshot-management.md:10]: Made the excerpt the first couple of sentences of the body

**Comment by kolchfa-aws** [comment]
@krisfreedain - I have incorporated editorial comments in the latest version of this blog.

**Comment by krisfreedain** [comment]
thanks @kolchfa-aws - safe for me to push this one live?

**Comment by kolchfa-aws** [comment]
@krisfreedain Sure. Let me know if you need anything from my side. Thanks!

File: _posts/snapshot-management.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+Snapshots are retried three times. Even if you don't set up notifications of failure in the SM policy, you can view the failure message using the [SM Explain API](https://opensearch.org/docs/latest/opensearch/snapshots/sm-api/#explain) or on the [policy details page in OpenSearch Dashboards](https://opensearch.org/docs/latest/opensearch/snapshots/sm-dashboards#view-edit-or-delete-an-sm-policy).
+
+### How does SM work under the hood?
+
+SM uses a state machine that has two workflows: creation and deletion. These workflows are executed sequentially, so SM does not start a new operation until the previous operation finishes. 
+
+The following image shows the creation workflow of the SM state machine.
+
+<img src="/assets/media/blog-images/2022-09-16/sm-state-machine.PNG" alt="SM State Machine" width="150" style="float: left; margin-right: 15px;"/>
```
Comment (krisfreedain) [inline _posts/snapshot-management.md:130]: please fix the path on this one - should be:
assets/media/blog-images/**2022-09-16-snapshot-management**/sm-state-machine.PNG

**Review by krisfreedain — CHANGES_REQUESTED** [review]
@kolchfa-aws - two changes required 

- change for the image path needed
- also please rename your main file from snapshot-management.md to 2022-09-16-snapshot-management.md

File: _posts/snapshot-management.md (hunk 2 above)
Comment (krisfreedain) [reply _posts/snapshot-management.md:130]: @kolchfa-aws - you will also need to create an author bio and image as part of this post. thanks

**Comment by kolchfa-aws** [comment]
@krisfreedain I fixed the image link and added author info. Also, I pushed the blog with a date in the name and without. Please feel free to delete or update if needed. Thanks!

**Comment by kolchfa-aws** [comment]
@krisfreedain I need to collaborate with @setiah on this blog. Please don't merge until I make changes. Thanks.

**Comment by krisfreedain** [comment]
> @krisfreedain I need to collaborate with @setiah on this blog. Please don't merge until I make changes. Thanks.

Sounds good. Please be sure to also complete your author bio file as well (kolchfa.markdown) - it's currently empty. Thanks

**Comment by setiah** [comment]
@kolchfa-aws apologies for the long wait. I have revised a few aspects of the blog to make it factually correct and a bit more descriptive. I have also added a demo gif. You can find my changes in the PR i raised on your remote branch. Once you merge them they will automatically reflect in here. https://github.com/kolchfa-aws/project-website/pull/1

**Comment by setiah** [comment]
> ### Description
> Adds a blog on Snapshot Management.
> 
> ### Issues Resolved
> none
> 
> ### Check List
> * [x ] Commits are signed per the DCO using --signoff
> 
> By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

pls add the issue link in the description - #987

File: _posts/2022-10-10-snapshot-management.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+A snapshot preserves two things: the cluster data and the cluster state. The cluster data includes OpenSearch index data, such as settings and mappings. The cluster state includes persistent cluster settings and index templates. Preserving the cluster state in a snapshot is optional and can be controlled with `include_global_state` parameter.
+
+### Why do you need to take snapshots?
+
+Let's go through the two most common scenarios in which you would need to take snapshots.
+
+**Scenario 1: Red Alert!** - Your cluster goes red because nodes went down, causing data loss. In this case, you can restore the red indexes from the most recent snapshot.
+
+**Scenario 2: Migration** - ** You need to migrate from one cluster to another. In this case, you can restore the indexes on your new cluster from a snapshot.
```
Comment (natebower) [inline _posts/2022-10-10-snapshot-management.md:36]:
```suggestion
**Scenario 2: Migration** – ** You need to migrate from one cluster to another. In this case, you can restore the indexes on your new cluster from a snapshot.
```

File: _posts/2022-10-10-snapshot-management.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+**Scenario 3: Restore to a point in time view** - You need to revert back to a point-in-time view. You can restore a snapshot to go back to the previous state.
```
Comment (natebower) [inline _posts/2022-10-10-snapshot-management.md:38]:
```suggestion
**Scenario 3: Restore to a point-in-time view** – You need to revert back to a point-in-time view. You can restore a snapshot to go back to the previous state.
```

Comment (natebower) [inline _posts/2022-10-10-snapshot-management.md:38]: Should it be "a previous state"?

File: _posts/2022-10-10-snapshot-management.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+**Scenario 4: Save costs** - You need to save historical data for compliance, but you primarily use only last seven days of data. You can save all previous data in a snapshot and delete it from your cluster to save cost.
```
Comment (natebower) [inline _posts/2022-10-10-snapshot-management.md:40]:
```suggestion
**Scenario 4: Save costs** – You need to save historical data for compliance, but you primarily use only the last seven days of data. You can save all previous data in a snapshot and delete it from your cluster to save costs.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Review complete.

File: _posts/2022-10-10-snapshot-management.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+      "failure": false,
+      "time_limit_exceeded": false
+    }
+  }
+}
+```
+
+You can also set up your SM policy through OpenSearch Dashboards as follows:
+
+<img src="/assets/media/blog-images/2022-10-10-snapshot-management/Snapshot-management-demo.gif" alt="SM Policy Create Workflow" width="150" style="float: left; margin-right: 15px;"/>
```
Comment (setiah) [inline _posts/2022-10-10-snapshot-management.md:129]: @kolchfa-aws @krisfreedain how do we check this gif is rendering properly on the website?

Comment (krisfreedain) [reply _posts/2022-10-10-snapshot-management.md:129]: @kolchfa-aws @setiah gif looks way too small:
Screen Shot 2022-10-07 at 10 47 55 AM


I recommend changing line 129 to:

`SM Policy Create Workflow{: .img-fluid }`

Comment (kolchfa-aws) [reply _posts/2022-10-10-snapshot-management.md:129]: Thank you, Kris! I will change that.

Comment (krisfreedain) [reply _posts/2022-10-10-snapshot-management.md]: Excellent - thanks @kolchfa-aws - I'll push this live on Monday morning.

**Pull request merged**